apps/vmq_swc/src/vmq_swc_store.erl

```diff
@@ -476,6 +477,23 @@ get_cached_node_clock(#swc_config{store=StoreName}) ->
     [{_, NodeClock}] = ets:lookup(StoreName, node_clock),
     NodeClock.
 
+maybe_get_cached_object(#swc_config{store=StoreName} = Config, SKey) ->
+    case ets:lookup(StoreName, SKey) of
+        [] ->
+            get_obj_for_key(Config, SKey);
+        [{_, Obj}] ->
+            Obj
+    end.
+
+r_o_w_cache_insert_object(#swc_config{store=StoreName}, SKey, Obj) ->
+    ets:insert(StoreName, {SKey, Obj}).
+
+r_o_w_cache_clear(#swc_config{store=StoreName}) ->
+    % the table max size is the size of a write batch, a select_delete shouldn't harm
+    % the overall performance
+    ets:select_delete(StoreName, [{{'$1', '_'}, [{'=/=', '$1', node_clock}], [true]}]),
```

Comment:
This is to me a sign that we should not use the same table. If we later decided to add something else to the table this would delete that something

Reply:
good point...